src/olympia/shelves/models.py

```diff
@@ -1,22 +1,28 @@
+from collections import namedtuple
+from urllib import parse
+
 from django.db import models
+from django.utils.functional import cached_property
 
 from olympia import amo
 from olympia.amo.models import ModelBase
-
-ENDPOINTS = ('collections', 'search')
-
-ENDPOINT_CHOICES = tuple((ty, ty) for ty in ENDPOINTS)
+from olympia.tags.models import Tag
 
 
 class Shelf(ModelBase):
+    Endpoints = namedtuple('Endpoints', ['COLLECTIONS', 'SEARCH', 'TAGS'])(
```

Comment:
Do you think we should call it `RANDOM_TAG` (or something like that), as opposed to `TAGS`, as the latter suggests to be that it might be used for getting a specific tag?

Reply:
I had that thought too but never got around to changing it.